common-custom-user-data-gradle-plugin/src/main/java/com/gradle/CommonCustomUserDataGradlePlugin.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * This Java source file was generated by the Gradle 'init' task.
+ */
+package com.gradle;
+
+import com.gradle.enterprise.gradleplugin.GradleEnterpriseExtension;
+import com.gradle.enterprise.gradleplugin.GradleEnterprisePlugin;
+import com.gradle.scan.plugin.BuildScanExtension;
+import com.gradle.scan.plugin.BuildScanPlugin;
+import org.gradle.api.Plugin;
+import org.gradle.api.Project;
+import org.gradle.api.initialization.Settings;
+
+public class CommonCustomUserDataGradlePlugin implements Plugin<Object> {
+    public void apply(Object target) {
+        if (target instanceof Settings) {
+            applySettingsPlugin((Settings) target);
+        } else if (target instanceof Project) {
+            applyProjectPlugin((Project) target);
+        }
+    }
+
+    private void applyProjectPlugin(Project project) {
+        project.getPlugins().withType(BuildScanPlugin.class, __ -> {
```

Comment:
What does __ do?

Reply:
This is a Kotlin-esque way to define a variable that is unused. I'm not sure if there's an idiomatic way to declare an unused variable for a Java lambda.